src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -503,7 +531,9 @@ async def get_repository_pex(
                     f"the resolve `{chosen_resolve.name}` (from `[python].resolves`)"
                 ),
                 resolve_name=chosen_resolve.name,
-                req_strings=request.requirements.req_strings,
+                # NB: PEX interprets `--lock` with no `req_strings` as "install the entire lockfile"
+                # And we don't use `req_strings` if the resolve isn't a PEX lockfile.
+                req_strings=FrozenOrderedSet(),
```

Comment:
I wish I did a better job naming this :( So confusing. Maybe we wait to rename things until we can delete requirement constraints?

Reply:
I concur